src/system_database.ts

```diff
@@ -185,8 +190,15 @@ export class PostgresSystemDatabase implements SystemDatabase {
         initStatus.applicationVersion,
         initStatus.applicationID,
         initStatus.createdAt,
+        initStatus.recovery,
       ]
     );
+    const workflow_retries = result.rows[0].workflow_retries;
+    if (workflow_retries >= initStatus.maxRetries && initStatus.recovery) {
```

Comment:
Could a user ever say "I want this to have infinite retries?"

Reply:
You could put a very large number in, I suppose, but I don't think we want to explicitly support infinite retries. The dead-letter queue is a good safeguard against bugs.